test/jigsaw-consumer/build.gradle

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+apply plugin: 'elasticsearch.build'
+
+sourceCompatibility = JavaVersion.VERSION_1_9
```

Comment:
The `test` folder is a collection of tools, and actual test projects live in `qa`, but for testing the build, I would move this to `buildSrc/src/testKit` and write a `GradleIntegrationTestCase` for it. 
The source and target comparability  can be parameters passed to the build so that we can test on newer than 9 also or control the Gradle version if we have too.